src/v2/providers/pubsub.ts

```diff
@@ -353,3 +356,66 @@ export function onMessagePublished<T = any>(
 
   return func;
 }
+
+/**
+ * @internal
+ *
+ * Adds a v1-style context to the event.
+ *
+ * @param event - The event to add the context to.
+ * @param topic - The topic the event is for.
+ */
+function attachPubSubContext<T>(event: CloudEvent<MessagePublishedData<T>>, topic: string) {
```

Comment:
i don't think we strictly need `topic`. all information should be derivable from event obj.